src/levenberg_marquardt.jl

```diff
@@ -17,25 +17,35 @@ Comp & Applied Math).
 * `tolX::Real=1e-8`: search tolerance in x
 * `tolG::Real=1e-12`: search tolerance in gradient
 * `maxIter::Integer=100`: maximum number of iterations
+* `min_step_quality=1e-3`: for steps below this quality, the trust region is shrinked
+* `good_step_quality=0.75`: for steps above this quality, the trust region is expanded
 * `lambda::Real=10.0`: (inverse of) initial trust region radius
+* `lambda_increase=10.`: `lambda` is multiplied by this factor after step below min quality
+* `lambda_decrease=0.1`: `lambda` is multiplied by this factor after good quality steps
 * `show_trace::Bool=false`: print a status summary on each iteration if true
 * `lower,upper=[]`: bound solution to these limits
 """
-function levenberg_marquardt{T}(f::Function, g::Function, initial_x::AbstractVector{T};
+function levenberg_marquardt{T}(f, g, initial_x::AbstractVector{T};
```

Comment:
Sure I can add it back in, but I am curious why? I always try to get rid of `::Function` since not all callables are `Function`.